### PR TITLE
Update Docker image and add freeze

### PR DESCRIPTION
**Pull request by ch-kr**
- updated default Docker image in `run_batches.py` to use image with up to date Hail version
- added warnings about hail version associated with Docker images
- added support for freeze in resource paths (need to remove rescue from paths in the future)
- removed merge with rescue search when finalizing RMC
- removed search_type from final RMC output
- updated flowchart link to remove rescue search
- removed references to rescue search

### CODE REVIEW

File: rmc/utils/constraint.py [hunk 1]
```diff
@@ -788,157 +768,15 @@ def get_rescue_1break_transcripts(
...
             search_num=1,
             is_break_found=False,
             is_breakpoint_only=False,
         )
     )
-    ht = ht.filter(~hl.literal(sections_with_breaks).contains(ht.section))
+    ht = ht.filter(~hl.literal(simul_sections).contains(ht.section))
     no_break_sections = ht.aggregate(hl.agg.collect_as_set(ht.section))
     logger.info(
         "%i transcripts did not have any evidence of RMC", len(no_break_sections)
```
Comment (lw453) [inline rmc/utils/constraint.py:782]: The `calculate_section_chisq` section at line 863 needs to be deleted, must have been a copy-paste mistake (not sure how that happened)

File: rmc/utils/mpc.py [hunk 2]
```diff
@@ -302,9 +280,8 @@ def create_context_with_oe(
     ht = ht.select(oe_col).key_by("locus", "alleles")
     ht = ht.collect_by_key()
     ht = ht.annotate(**{oe_col: hl.nanmin(ht.values[oe_col])})
-    transcript_col = "transcript_rescue" if include_rescue else "transcript"
     ht = ht.annotate(
-        **{transcript_col: ht.values.find(lambda x: x[oe_col] == ht[oe_col]).transcript}
+        **{"transcript": ht.values.find(lambda x: x[oe_col] == ht[oe_col]).transcript}
```
Comment (lw453) [inline rmc/utils/mpc.py:284]:
```suggestion
    ht = ht.select("oe").key_by("locus", "alleles")
    ht = ht.collect_by_key()
    ht = ht.annotate(**{"oe": hl.nanmin(ht.values["oe"])})
    ht = ht.annotate(
        **{"transcript": ht.values.find(lambda x: x["oe"] == ht["oe"]).transcript}
    )

File: rmc/utils/mpc.py [hunk 3]
```diff
@@ -302,9 +280,8 @@ def create_context_with_oe(
...
     )
     ht = ht.drop("values")
```
Comment (lw453) [inline rmc/utils/mpc.py:287]: Can't comment but the section on lines 288-302 hasn't yet been edited to remove the joining code from the rescue PR:

File: rmc/utils/mpc.py [hunk 4]
```diff
@@ -405,16 +378,12 @@ def prepare_pop_path_ht(
...
     # in canonical transcripts does not exist, create it
     # If HT exists but without specified version of OE annotation, add that OE annotation
-    rmc_cols = {
-        "transcript": "transcript_rescue" if include_rescue else "transcript",
-        "oe": "oe_rescue" if include_rescue else "oe",
-    }
+    rmc_cols = {"transcript", "oe"}
     if not file_exists(context_with_oe_dedup.path) or (
         len(set(rmc_cols.values()).intersection(context_with_oe_dedup.ht().row))
         < len(set(rmc_cols.values()))
```
Comment (lw453) [inline rmc/utils/mpc.py:384]:
```suggestion
    if not file_exists(context_with_oe_dedup.path) or (
        len(rmc_cols).intersection(context_with_oe_dedup.ht().row))
        < len(rmc_cols)
```

File: rmc/utils/mpc.py [hunk 5]
```diff
@@ -405,16 +378,12 @@ def prepare_pop_path_ht(
...
     ):
         create_context_with_oe(
-            include_rescue=include_rescue,
             overwrite_temp=overwrite_temp,
             overwrite_output=overwrite_output,
         )
```
Comment (lw453) [inline rmc/utils/mpc.py:389]: Can't comment but lines 393-397 should be deleted

**Review by lw453 — CHANGES_REQUESTED** [review]

**Comment by lw453** [comment]
Also reference to the "initial" pathway can be removed in `constraint.py` line 938

File: rmc/utils/constraint.py (hunk 1 above)
Comment (ch-kr) [reply rmc/utils/constraint.py:782]: closing this since you've deleted the extra function

File: rmc/utils/mpc.py (hunk 2 above)
Comment (ch-kr) [reply rmc/utils/mpc.py:284]: am I missing something? this looks right, right?
image

File: rmc/utils/mpc.py (hunk 5 above)
Comment (ch-kr) [reply rmc/utils/mpc.py:389]: image
this part? that's just a checkpoint though, right?

File: rmc/utils/mpc.py (hunk 2 above)
Comment (ch-kr) [reply rmc/utils/mpc.py:284]: ohh, oe_col. will fix

**Comment by lw453** [comment]
I went through the repo again, a few things unrelated to this PR:

Line 93 in `regional_constraint.py` is broken, I think it should look something like
```
            exome_ht = exome_ht.select(
                ac=exome_ht.freq[0].AC,
                af=exome_ht.freq[0].AF,
                pass_filters=exome_ht.pass_filters,
                filters=exome_ht.filters,
                exome_coverage=exome_ht.coverage.exomes.median,
                transcript_consequences=exome_ht.transcript_consequences,
            )
            exome_ht = exome_ht.filter(
                keep_criteria(exome_ht, exome_ht.ac, exome_ht.af, exome_ht.filters)
            )
```

Line 247 and 316 in `generic.py`:
```
ht = filter_vep_to_canonical_transcripts(ht, vep_root="vep")
```

`TEMP_PATH_WITH_SLOW_DEL` is unused import in `missense_badness.py`

**Review by lw453 — APPROVED** [review]

**Pull request merged**